### DISCUSSION TITLE
Connection Terminated

### DISCUSSION THREAD

**Discussion by Silas888** [LizardByte/community#112] [Sunshine] [answered]
Hi everyone. I'm new to Sunshine but not to Moonlight. Previously I had Moonlight connected via GameStreaming with no issues but like many are forced to find a new method in light of Nvida's announcement. When connected to Sunshine, I'm coming across issues where when launching a game (added as an application in Sunshine) Moonlight will give me a "Connection Terminated" message however the game will properly launch on the PC. Specifically for me the games have been Halo Master Chief Collection. For game like Arkham Knights and God of War, I get a straight "Connection Error" and NOT launch the game at all. Mind you these are all games that I purchased and download. They are not Steam games. Needless to say all of the games worked flawlessly via GameStream. I'm curious if anyone has come across this? I'm hoping its a configuration issue. Thanks for you help.

Sunshine commit or version
v0.16.0

GPU Type
Intel I7-8700

GPU Model
Nvida GeForce GTX 1080

**Comment by KuleRucket** [comment, marked as answer]
If you are running sunshine as a service, stop the service and run from a window. The service launches applications as SYSTEM user and not the logged in user (known issue) so the application you are running might not be looking in the correct place for its configuration. You can also try to get the service to run as the local user.

See also:
https://lizardbyte--574.org.readthedocs.build/projects/sunshine/en/574/about/usage.html#application-list

**Reply by Silas888** [reply]
Thank you for your reply. No I'm running Sunshine directly from the desktop as a local user. I have seen the commands that you sent, and for the most part understand them. It looks like you can use variables however, there is no example of how the variables are used so I can't figure out how that works or if that is what it even means.

**Reply by dom191984** [reply]
Did you manage to get this to work? I am having exactly the same issue

**Reply by Silas888** [reply]
Yes. I created a batch file and placed in the game directly. I referenced the game EXE from the batch file, and then referenced the batch file from Sunshine.

**Reply by dom191984** [reply]
Ok so you launch via the batch file? that seems long winded. Is it easy to create/write?

**Reply by StellarJay77** [reply]
I can launch sunshinesvc as the local logged in user but then the http://localhost:47990/ doesn't work. As soon as I run it as the SYSTEM user it works but I get Connection Terminated again. Is there any way to run Sunshine as a service in Windows and have it work?

**Reply by dom191984** [reply]
It's cool worked out what the issue was, sorted now. Thanks everyone

**Reply by StellarJay77** [reply]
how did you fix it? I'm having the same issue.

**Reply by dom191984** [reply]
It ended up being that I had not allowed the correct firewall settings for the sunshine.exe in Windows defender firewall.

This is my suggestion and test steps
- First go to the firewall by searching "Windows defender firewall" in the main windows search
- On the left hand side I then click on the option "Turn off window defender on or off settings"
- I then turn off for testing purposes the both the public and private network firewalls and then tested the connection
- I then noticed that it connected fine. I turned them back on immediately after. 
Please note the obvious risks turning the firewall and do this at you own risk. As I said, I did this to test.

Next step to fix or how I fixed it
In the  Windows defender firewall where you are do the following
- on the left hand side now click on "Advanced settings"
- In advanced settings click on inbound rules
- In inbound rules search for sunshine by scrolling down to it
- I found that I had not allowed the access so had a number of accesses for Sunshine with a red circle and a slash threw it
- I click and then deleted these and it worked

I hope that helps?

Any issues please drop us a message
Cheers

**Reply by dom191984** [reply]
Did this work?

**Comment by SokarCodes** [comment]
I have similar behaviour on my Windows10 machine (RTX4090, AMD 7600x CPU). Gamestream with Geforce Experience works flawlessly on my AppleTV Moonlight client but with Sunshine I get instant Connection Terminated when launching games. I attach log of my failed launch of Horizon Zero Dawn game. It is captured with verbose flag if it helps at all.

[failed-horizon-zero-dawn-launch.txt](https://github.com/LizardByte/Sunshine/files/10269040/failed-horizon-zero-dawn-launch.txt)

**Reply by Silas888** [reply]
Thank you. Unfortunately, it doesn't help me but maybe it might have somebody else's more adept than me. Funny thing though. Even though I set up a log when I get an error, nothing is written to it.

**Reply by SokarCodes** [reply]
You can download sunshine as zip package and run it as administrator. It opens terminal with all the logging data and you can just copy output of that terminal window to text editor for example.

**Comment by skarabaraks** [comment]
worth checking that when launching Playnite (Playnite.DesktopApp.exe) it doesn't automatically switch to the fullscreen mode of Playnite (Playnite.FullscreenApp.exe)
